content/03_CreatingObservableSequences.md

```diff
@@ -0,0 +1,1274 @@
+
+Nobody should be using the APM today, but for completeness (and just in case you have to use an ancient library that only offers the APM) I will provide a very brief explanation of Rx's support for it.
+
+The result of the call to `Observable.FromAsyncPattern` does _not_ return an observable sequence. It returns a delegate that returns an observable sequence. (So it is essentially a factory factory) The signature for this delegate will match the generic arguments of the call to `FromAsyncPattern`, except that the return type will be wrapped in an observable sequence. The following example wraps the `Stream` class's `BeginRead`/`EndRead` methods (which are an implementation of the APM).
+
+**Note**: this is purely to illustrate how to wrap the APM. You would never do this in practice because `Stream` has supported the TAP for years.
+
+```csharp
+Stream stream = GetStreamFromSomewhere();
+var fileLength = (int) stream.Length;
+
+Func<byte[], int, int, IObservable<int>> read = Observable.FromAsyncPattern<byte[], int, int, int>(
+    stream.BeginRead, 
+    stream.EndRead);
+var buffer = new byte[fileLength];
+IObservable<int> bytesReadStream = read(buffer, 0, fileLength);
+bytesReadStream.Subscribe(byteCount =>
+{
+    Console.WriteLine("Number of bytes read={0}, buffer should be populated with data now.", byteCount);
+});
+```
+
+## Subjects
+
+So far, this chapter has explored various factory methods that return `IObservable<T>` implementations. There is another way though: `System.Reactive` defines various types that implement `IObservable<T>` that we can instantiate directly. But how do we determine what values these types produce? We're able to do that because they also implement `IObserver<T>`, enabling us to push values into them, and those very same values we push in will be the ones seen by observers.
+
+Types that implement both `IObservable<T>` and `IObserver<T>` are called _subjects_ in Rx. There's an an `ISubject<T>` to represent this. (This is in the `System.Reactive` NuGet package, unlike `IObservable<T>` and `IObserver<T>`, which are both built into the .NET runtime libraries.) `ISubject<T>` looks like this:
```

Comment:
I think a picture/diagram might help here for people to visualise what's going on. Events emerge from "inside" the things we've been creating before, rather than being shoved in from outside.

Reply:
I've been struggling to envisage what you have in mind.